src/main/java/net/pms/util/SubtitleColor.java

```diff
@@ -0,0 +1,475 @@
+					sb.append("0").append(red);
+				} else {
+					sb.append(red);
+				}
+				i++;
+			} else if ("R".equals(pattern.substring(i, i+1))) {
+				sb.append(red);
+			} else if ("GG".equals(pattern.substring(i, i+2))) {
+				if (green.length() < 2) {
+					sb.append("0").append(green);
+				} else {
+					sb.append(green);
+				}
+				i++;
+			} else if ("G".equals(pattern.substring(i, i+1))) {
+				sb.append(green);
+			} else if ("BB".equals(pattern.substring(i, i+2))) {
+				if (blue.length() < 2) {
+					sb.append("0").append(blue);
+				} else {
+					sb.append(blue);
+				}
+				i++;
+			} else if ("B".equals(pattern.substring(i, i+1))) {
+				sb.append(blue);
+			} else if ("AA".equals(pattern.substring(i, i+2))) {
+				if (alpha.length() < 2) {
+					sb.append("0").append(alpha);
+				} else {
+					sb.append(alpha);
+				}
+				i++;
+			} else if ("A".equals(pattern.substring(i, i+1))) {
+				sb.append(alpha);
+			}
+		}
+
+		if (suffix != null) {
+			sb.append(suffix);
+		}
+		return sb.toString();
+	}
+
+	@Override
+    public String toString() {
+        return
+        	getClass().getSimpleName() +
+        	"[r=" + getHexValue("0x", "RR", null, true, false) +
+        	", g=" + getHexValue("0x", "GG", null, true, false) +
+        	", b=" + getHexValue("0x", "BB", null, true, false) +
+        	", a=" + getHexValue("0x", "AA", null, true, false) +
+        	"]";
+    }
+
+	protected static String extractHexString(String s) {
+		Matcher matcher = hexPattern.matcher(s);
+		if (matcher.find()) {
+			return matcher.group(1).toUpperCase(Locale.ROOT);
+		} else {
+			return null;
+		}
+	}
+
+	protected static String extractDecString(String s) {
+		Matcher matcher = decPattern.matcher(s);
+		if (matcher.find()) {
+			return matcher.group(1);
+		} else {
+			return null;
+		}
+	}
+
+	/**
+	 * Try to create a {@link SubtitleColor} from {@code subtitleColor}. The
+	 * parsing will try 4 different parsing methods:
+	 *
+	 * <ul>
+	 * <li>First try to parse {@code subtitleColor} as a hexadecimal string.
+	 * Several different notations are supported and it can be with or without
+	 * alpha. Valid forms are RRGGBB or RRGGBBAA with one of the following
+	 * prefixes: "x'", "%", "#", "0x", "\x", "%x", "$", "h'", "16#", "16r",
+	 * "#x", "#16r", "&amp;h" or "0h".</li>
+	 * <li>If that fails, try to parse {@code subtitleColor} as a decimal string
+	 * without alpha.</li>
+	 * <li>If that also fails, try to parse {@code subtitleColor} as a color
+	 * name (Red, Yellow etc).</li>
+	 * <li>As a last attempt, try to parse {@code subtitleColor} as a system
+	 * property integer.</li>
+	 * </ul>
+	 * If the parsing fails, {@code null} is returned.
+	 *
+	 * @param subtitleColor the {@link String} to attempt to parse a
+	 *            {@link SubtitleColor} from.
+	 * @return A newly created {@link SubtitleColor} or {@code null} it the
+	 *         parsing fails.
+	 */
+	public static SubtitleColor toSubtitleColor(String subtitleColor) {
```

Comment:
@valib I took a look at that implementation, but it doesn't quite do the same (although almost). It's not used for subtitle color but for color in general and thus returns a ```Color``` instead of a ```SubtitleColor```. It would certainly be possible to merge the two into one, but the implementation looks ok and I think they can stay separate for now.